src/components/AxiosInterceptor/helpers/useSetNativeAuthInterceptors.ts

```diff
@@ -0,0 +1,80 @@
+import axios from 'axios';
+import { useSelector } from 'react-redux';
+import { retry } from 'helpers';
+import { useAxiosInterceptorContext } from 'lib';
+import { IS_TEST } from 'localConstants';
+import { accountSelector } from 'redux/selectors';
+
+let nativeAuthToken: string | undefined;
+let requestInterceptorId = -1;
+
+export const useSetNativeAuthInterceptors = () => {
+  const {
+    loginInfo: { tokenLogin }
+  } = useAxiosInterceptorContext();
+  const { externalNativeAuthToken } = useSelector(accountSelector);
+  nativeAuthToken = externalNativeAuthToken || tokenLogin?.nativeAuthToken;
+
+  /**
+   * `isLoggedIn` gets set before token arrives, so this polling makes
+   * `axios` wait until token arrives
+   */
+  const getToken = async (): Promise<string> => {
+    const pollForToken = () => {
```

Comment:
This could be an implicit `return`.

Reply:
There is a `return retry(pollForkToken` not `return pollForToken`